flytekit/remote/remote.py

```diff
@@ -738,42 +770,24 @@ def _serialize_and_register(
         # Create dummy serialization settings for now.
         # TODO: Clean this up by using lazy usage of serialization settings in translator.py
         serialization_settings = settings
-        is_dummy_serialization_setting = False
         if not settings:
             serialization_settings = SerializationSettings(
                 ImageConfig.auto_default_image(),
                 project=self.default_project,
                 domain=self.default_domain,
                 version=version,
             )
-            is_dummy_serialization_setting = True
-
         if serialization_settings.version is None:
             serialization_settings.version = version
 
         _ = get_serializable(m, settings=serialization_settings, entity=entity, options=options)
-
-        ident = None
-        for entity, cp_entity in m.items():
-            if not isinstance(cp_entity, admin_workflow_models.WorkflowSpec) and is_dummy_serialization_setting:
-                # Only in the case of workflows can we use the dummy serialization settings.
-                raise user_exceptions.FlyteValueException(
-                    settings,
-                    f"No serialization settings set, but workflow contains entities that need to be registered. {cp_entity.id.name}",
-                )
-
-            try:
-                ident = self.raw_register(
-                    cp_entity,
-                    settings=settings,
-                    version=version,
-                    create_default_launchplan=create_default_launchplan,
-                    options=options,
-                    og_entity=entity,
-                )
-            except RegistrationSkipped:
-                pass
-
+        # concurrent register
+        cp_task_entity_map = OrderedDict(filter(lambda x: isinstance(x[1], task_models.TaskSpec), m.items()))
+        ident = asyncio.run(self._register(cp_task_entity_map, settings, version))
+        # serial register
+        cp_other_entities = OrderedDict(filter(lambda x: not isinstance(x[1], task_models.TaskSpec), m.items()))
+        for entity, cp_entity in cp_other_entities.items():
+            ident = self.raw_register(cp_entity, settings, version, og_entity=entity)
```

Comment:
Could we use `loop.run_in_executor(None, functools.partial(self.raw_register, ..))` here instead? so we don't need to create another function `async def _register()`

Reply:
sure thing